exercises/bowling/canonical-data.json

```diff
@@ -1,117 +1,110 @@
+      "description": "rolling a spare with the two roll bonus does not get a bonus roll",
+      "rolls": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 10, 7, 3],
+      "expected": 20
     }, {
-      "description": "should allow fill balls to be a strike",
+      "description": "strikes with the two roll bonus do not get bonus rolls",
       "rolls": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 10, 10, 10],
       "expected": 30
     }, {
-      "description": "should be able to score a perfect game",
+      "description": "a strike with the one roll bonus after a spare in the last frame does not get a bonus",
+      "rolls": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 7, 3, 10],
+      "expected": 20
+    }, {
+      "description": "all strikes is a perfect game",
       "rolls": [10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10],
       "expected": 300
-    }]
-  },
-  "validations": {
-    "description": [
-      "Check game rules."
-    ],
-    "cases": [{
-      "description": "should not allow rolls with negative pins",
-      "rolls": [-1],
-      "expected": "Pins must have a value from 0 to 10"
     }, {
-      "description": "should not allow rolls better than strike",
-      "rolls": [11],
-      "expected": "Pins must have a value from 0 to 10"
+      "description": "Rolls can not score negative points",
+      "rolls": [-1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+      "expected": -1
     }, {
-      "description": "should not allow two normal rolls better than strike",
-      "rolls": [5, 6],
-      "expected": "Pin count exceeds pins on the lane"
+      "description": "A roll can not score more than 10 points",
+      "rolls": [11, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+      "expected": -1
     }, {
-      "description": "should not allow two normal rolls better than strike in last frame",
-      "rolls": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 5, 6],
-      "expected": "Pin count exceeds pins on the lane"
+      "description": "Two rolls in a frame can not score more than 10 points",
+      "rolls": [5, 6, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+      "expected": -1
     }, {
-      "description": "should not allow to take score at the beginning of the game",
+      "description": "An unstarted game can not be scored",
       "rolls": [],
-      "expected": "Score cannot be taken until the end of the game"
+      "expected": -1
     }, {
-      "description": "should not allow to take score before the game has ended",
-      "rolls": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-      "expected": "Score cannot be taken until the end of the game"
+      "description": "An incomplete game can not be scored",
+      "rolls": [0, 0],
+      "expected": -1
     }, {
-      "description": "should not allow rolls after the tenth frame",
+      "description": "A game with more than ten frames can not be scored",
```

Comment:
For those of you who aren't Peter, we've been talking about this in the Rust PR as well. 

https://github.com/exercism/xrust/pull/213

For Rust, at least, we're going to have the `roll` method fail if the roll is invalid.